HerPublicWebsite/Models/Questionnaire/ReviewEpcViewModel.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using GovUkDesignSystem.Attributes.ValidationAttributes;
+using HerPublicWebsite.BusinessLogic.Models;
+using HerPublicWebsite.Extensions;
+
+namespace HerPublicWebsite.Models.Questionnaire;
+
+public class ReviewEpcViewModel : QuestionFlowViewModel
+{
+    public enum YesOrNo
+    {
+        Yes,
+        No
+    }
+
+    [GovUkValidateRequired(ErrorMessageIfMissing = "Select whether this EPC is correct for you property")]
+    public YesOrNo? EpcIsCorrect {get; set;}
+    
+    public string AddressRawHtml { get; set; }
+    public string EpcBand { get; set; }
+    public string ValidFrom { get; set; }
+    public string ValidUntil { get; set; }
+    public string PropertyType { get; set; }
+
+    public ReviewEpcViewModel() {}
+    
+    public ReviewEpcViewModel(EpcDetails epcDetails, bool? epcIsCorrect)
+    {
+        var addressParts = new List<string>
+            {
+                epcDetails.AddressLine1,
+                epcDetails.AddressLine2,
+                epcDetails.AddressLine3,
+                epcDetails.AddressLine4,
+                epcDetails.AddressTown,
+                epcDetails.AddressPostcode
+            }
+            .Where(s => !string.IsNullOrWhiteSpace(s))
+            .Select(WebUtility.HtmlEncode);
+
+        AddressRawHtml = string.Join("<br/>", addressParts);
+        EpcBand = epcDetails.EpcRating.ToString();
+        ValidFrom = epcDetails.LodgementDate != null ? epcDetails.LodgementDate.Value.ToString("dd MMMM yyyy") : "Unknown";
+        ValidUntil = epcDetails.ExpiryDate != null ? epcDetails.ExpiryDate.Value.ToString("dd MMMM yyyy") : "Unknown";
+
+        PropertyType =
+            (epcDetails.HouseType?.ForDisplay() ??
+             epcDetails.BungalowType?.ForDisplay() ??
+             epcDetails.FlatType?.ForDisplay() ??
+             "") + " " + (epcDetails.PropertyType?.ForDisplay() ?? "Unknown");
```

Comment:
Potentially worth a refactor, but I'm undecided on whether or not I'm happy with chaining null coalescing operators like this (although the review I'll be putting up for you later has a nested ternary operator so I'm probably not one to talk)

Reply:
It's a bit yucky I agree, I'll have another look at it and see if I can neaten it up without making it huge.